utils/python/CIME/XML/env_batch.py

```diff
@@ -249,7 +249,7 @@ def set_job_defaults(self, bjobs, pesize=None):
                 task_count = int(task_count)
             queue = self.select_best_queue(task_count)
             self.set_value("JOB_QUEUE", queue, subgroup=job)
-            walltime = self.get_max_walltime(queue)
+            walltime = self.get_max_walltime(queue) if walltime is None else walltime
```

Comment:
Not clear to me when `walltime is None` will actually be true. Is this based on `os.environ['KEY']` being `None` when `KEY` is not in the environment?